core/modules/xrdsvc/plugin.cc

```diff
@@ -20,30 +20,57 @@
  * the GNU General Public License along with this program.  If not,
  * see <http://www.lsstcorp.org/LegalNotices/>.
  */
-/// Implement XrdSsiGetServerService() to provide Qserv's SsiService
+/// Implement XrdSsiProviderServer to provide Qserv's SsiService
 /// implementation. Link this file when building a plugin to be used as
-/// ssi.svclib .
-
+/// ssi.svclib.
 
 // System headers
-#include <iostream>
+#include <memory>
 
 // Third-party headers
-#include "XrdSsi/XrdSsiLogger.hh"
-#include "XrdSsi/XrdSsiService.hh"
+#include "XrdSsi/XrdSsiProvider.hh"
 
 // Qserv headers
-#include "xrdsvc/SsiService.h"
+#include "SsiService.h"
 
-class XrdSsiLogger;
-class XrdSsiCluster;
+namespace lsst {
+namespace qserv {
+namespace xrdsvc {
 
-extern "C" {
-XrdSsiService *XrdSsiGetServerService(XrdSsiLogger  *logP,
-                                      XrdSsiCluster *clsP,
-                                      const char    *cfgFn,
-                                      const char    *parms)
+class SsiProviderServer : public XrdSsiProvider
 {
-    return new lsst::qserv::xrdsvc::SsiService(logP);
-}
-} // extern "C"
+public:
+
+    virtual XrdSsiService *GetService(XrdSsiErrInfo& eInfo, char const* contact, int oHold=256)
+    {
+        return _service.get();
+    }
+
+    virtual bool Init(
+        XrdSsiLogger* logP,
+        XrdSsiCluster* clsP,
+        char const* cfgFn,
+        char const* parms,
+        int argc,
+        char **argv)
+    {
```

Comment:
Perhaps, collapse the arguments down to one or two lines as we don't seem to be using most of them.